File: solutions/elixir/darts/2/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    euclidean_distance = :math.sqrt(x*x + y*y)

    cond do
      euclidean_distance <= 1 -> 10
      euclidean_distance <= 5 -> 5
      euclidean_distance <= 10 -> 1
      true -> 0
    end
  end
end

File: solutions/typescript/darts/2/darts.ts
export function score(x: number, y: number): number {
  const euclidean_distance = Math.hypot(x, y)

  if (euclidean_distance <= 1) return 10
  if (euclidean_distance <= 5) return 5
  if (euclidean_distance <= 10) return 1
  return 0
}
